main.cpp

```diff
@@ -154,6 +161,13 @@ int addFile(char* name, const char* path)
         std::cout << "file size: " << size << std::endl;
     }
 
+    // read and update file modification time
+    #if CONFIG_SPIFFS_USE_MTIME
+    struct stat sbuff;
+    stat(path, &sbuff);
+    SPIFFS_fupdate_meta(&s_fs, dst, &sbuff.st_mtime);
```

Comment:
Can you please handle the possibly different `sizeof(time_t)` explicitly? E.g.

```
uint32_t meta = (maybe truncate?) sbuff.st_mtime;
SPIFFS_fupdate_meta(&s_fs, dst, &meta);
```

And similar in unpackFile below.